src/ClientData/include/ClientData/CallstackEvent.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CLIENT_DATA_CALLSTACK_EVENT_H_
+#define CLIENT_DATA_CALLSTACK_EVENT_H_
+
+#include <stdint.h>
+
+namespace orbit_client_data {
+
+class CallstackEvent {
+ public:
+  CallstackEvent() = delete;
+  CallstackEvent(uint64_t timestamp_ns, uint64_t callstack_id, uint32_t thread_id)
```

Comment:
I was thinking: `CallstackSample` in `capture.proto` follows the order `tid`, `callstack_id`, `timestamp_ns`. What do you prefer, if you have any opinion at all?

Reply:
I feel you, switching from the setters to the constructor calls required care. Do you think it was better with the setters? I wanted to get rid of them to make the type immutable and its construction a single line, but I'd like to hear what you think.

Reply:
Yes, setters would solve that particular problem. But I agree with you: The immutability is more important here.